src/theory/bv/slicer.cpp

```diff
@@ -20,13 +20,35 @@
 #include "theory/bv/theory_bv_utils.h"
 #include "theory/rewriter.h"
 
-using namespace CVC4;
-using namespace CVC4::theory;
-using namespace CVC4::theory::bv;
 using namespace std; 
 
 
-const TermId CVC4::theory::bv::UndefinedId = -1; 
+namespace CVC4 {
+namespace theory {
+namespace bv {
+
+const TermId UndefinedId = -1;
+
+namespace {
+
+void intersect(const std::vector<TermId>& v1,
+               const std::vector<TermId>& v2,
+               std::vector<TermId>& intersection)
+{
+  for (const TermId id1 : v1)
+  {
+    for (const TermId id2 : v2)
+    {
+      if (v2[id2] == v1[id1])
```

Comment:
Shouldn't this be `id2 == id1`?

Reply:
Good catch. Interestingly, when I tested the new against the old implementation, no discrepancy was encountered.

Reply:
Maybe I forgot to enable the slicer, that would be the only explanation for that I can think of...